src/java.base/share/classes/java/lang/reflect/AccessibleObject.java

```diff
@@ -520,12 +521,16 @@ protected AccessibleObject() {}
      * <p> Note that any annotation returned by this method is a
      * declaration annotation.
      *
+     * @implSpec
+     * The default implementation throws {@code
+     * IllegalStateException}; subclasses should override this method.
+     *
      * @throws NullPointerException {@inheritDoc}
      * @since 1.5
      */
     @Override
     public <T extends Annotation> T getAnnotation(Class<T> annotationClass) {
-        throw new AssertionError("All subclasses should override this method");
+        throw new IllegalStateException("All subclasses should override this method");
```

Comment:
I'm now sure that ISE is the more appropriate exception here because there isn't an alternative state that would accept the input. UOE might be better.

Reply:
I'm not sure that ISE is the most appropriate exception here because there isn't an alternative state that would accept the input. UOE might be better.

The new proposal to just seal Executable looks reasonable.